internal/v5/resources.go

```diff
@@ -161,11 +138,63 @@ func (h *ReqHandler) metaResources(entity *mongodoc.Entity, id *router.ResolvedU
 	return results, nil
 }
 
+// GET id/meta/resource
+// https://github.com/juju/charmstore/blob/v5/docs/API.md#get-idmetaresources
+func (h *ReqHandler) metaResourcesSingle(entity *mongodoc.Entity, id *router.ResolvedURL, path string, flags url.Values, req *http.Request) (interface{}, error) {
+	data, err := h.metaResourcesSingle0(entity, id, path, flags, req)
+	if err != nil {
+		if errgo.Cause(err) == params.ErrNotFound {
+			logger.Infof("replacing not-found error on %s/meta/resources%s: %v (%#v)", id.URL.Path(), path, err, err)
+			// It's a not-found error; return nothing
+			// so that it's OK to use this in a bulk meta request.
+			return nil, nil
+		}
+		return nil, errgo.Mask(err)
+	}
+	return data, nil
+}
+
+func (h *ReqHandler) metaResourcesSingle0(entity *mongodoc.Entity, id *router.ResolvedURL, path string, flags url.Values, req *http.Request) (interface{}, error) {
+	if id.URL.Series == "bundle" {
+		return nil, nil
+	}
+	rid, err := parseResourceId(strings.TrimPrefix(path, "/"))
+	if err != nil {
+		return nil, errgo.WithCausef(err, params.ErrNotFound, "")
+	}
+	ch, err := h.entityChannel(id)
+	if err != nil {
+		return nil, errgo.Mask(err)
+	}
+	doc, err := h.Store.ResolveResource(id, rid.Name, rid.Revision, ch)
+	if err != nil {
+		if errgo.Cause(err) != params.ErrNotFound || rid.Revision != -1 {
+			return nil, errgo.Mask(err, errgo.Is(params.ErrNotFound))
+		}
+		// The resource wasn't found and we're not asking for a specific
+		// revision. If the resource actually exists in the charm metadata,
+		// return a placeholder document as would be returned by
+		// the /meta/resources (ListResources) endpoint.
+		if _, ok := entity.CharmMeta.Resources[rid.Name]; !ok {
+			return nil, errgo.Mask(err, errgo.Is(params.ErrNotFound))
+		}
+		doc = &mongodoc.Resource{
```

Comment:
In fact the only thing that matters is the resource name, and that doesn't really feel that crucial to keep in sync to me, as it's a local matter between this code and fromResourceDoc. I'll maybe lose the other two fields.